contrib/win32/win32compat/misc.c

```diff
@@ -1044,58 +1044,62 @@ w32_strerror(int errnum)
 		return _sys_errlist_ext[errnum - EADDRINUSE];
 	return strerror(errnum);
 }
-/* 
- * Temporary implementation of readpassphrase. 
- * TODO - this needs to be reimplemented as per 
- * https://linux.die.net/man/3/readpassphrase
- */
-char * 
-readpassphrase(const char *prompt, char *out, size_t out_len, int flags) {
-	char *askpass = NULL;
-	char *ret = NULL;
 
-	DWORD mode;
-	size_t len = 0;
-	int retr = 0;
+char *
+readpassphrase(const char *prompt, char *outBuf, size_t outBufLen, int flags) {
```

Comment:
Unix has the concept of reading from /dev/tty.. 
if we fail to open /dev/tty and RPP_REQUIRE_TTY is set to false then unix reads the input from STDIN..

since in windows there is no concept of /dev/tty this flag and the corresponding logic is ignored..